ui-tests/src/test/java/io/syndesis/qe/pages/integrations/editor/add/steps/DataMapper.java

```diff
@@ -174,14 +174,50 @@ public void selectMapping(String mappingName, SelenideElement containerElement)
         List<String> path = Arrays.asList(mappingName.trim().split("\\."));
         final String parent = path.get(0);
 
-        path.forEach(s -> {
-            SelenideElement detailElement = containerElement.find(By.id(s)).shouldBe(visible);
+        boolean previousHadNullId = false;
+
+        for( int i = 0; i < path.size(); i++ ) {
+            String s = path.get(i);
+            SelenideElement detailElement = containerElement.find(By.id(s));
+
+            if (s == null || s.equals("")) {
+                log.info("WWWWW ID is null");
+                previousHadNullId = true;
+                continue;
+            } else {
+                if (previousHadNullId) {
+                    log.info("WWWWW previous ID was null");
+                    log.info("WWWWW current  ID is  " + s);
+
+                    if (!detailElement.is(visible)) {
+                        log.info("WWWWW clicking an ancestor");
+                        containerElement.find(By.cssSelector("div[class=\"DocumentFieldDetailComponent\"")).shouldBe(visible).click();
+                    } else {
+                        //go on
+                    }
+                }
+                if(detailElement.exists()) {
+                    detailElement.shouldBe(visible);
+                } else {
+                    containerElement.find(By.cssSelector("div[class=\"DocumentFieldDetailComponent\"")).shouldBe(visible).click();
+                    detailElement.shouldBe(visible);
+                }
+            }
+
             if (detailElement.find(Element.CHILDREN).exists()) {
```

Comment:
I know this is not your code but since we are touching it already, could you flip the condition here to prevent empty `if`?

Reply:
Another good practice is having positive conditions in Ifs. I'd leave it as it is. It's well readable.